packages/solana/src/__test__/schemas.test.ts

```diff
@@ -0,0 +1,102 @@
+import * as base58 from '@chainflip/utils/base58';
+import { assetContractId, chainContractId } from '@chainflip/utils/chainflip';
+import * as ss58 from '@chainflip/utils/ss58';
+import { BN } from '@coral-xyz/anchor';
+import { describe, expect, it } from 'vitest';
+import { type z } from 'zod';
+import { swapSchema } from '../schemas';
+
+describe('swapSchema', () => {
+  it.each([
+    ['Bitcoin', assetContractId.Btc, 'bc1qqt3juqef9azhd0zeuamu9c30pg5xdllvmks2ja'],
+    [
+      'Polkadot',
+      assetContractId.Dot,
+      ss58.decode('1BzDB5n2rfSJwvuCW9deKY9XnUyys8Gy44SoX8tRNDCFBhx').data,
+    ],
+    [
+      'Solana',
+      assetContractId.SolUsdc,
+      base58.decode('3yKDHJgzS2GbZB9qruoadRYtq8597HZifnRju7fHpdRC'),
+    ],
+  ] as const)('handles %s destinations', (chain, asset, address) => {
```

Comment:
maybe add an eth example

Reply:
The ETH examples are all in `deposit.test.ts`. These tests are here to increase the coverage

Reply:
Although I ended up adding more tests there anyway eventually